scripts/extensions/auto-tagging-widget/src/ImageTaggingComponent/ImageTaggingComponent.tsx

```diff
@@ -0,0 +1,355 @@
+        this._mounted &&
+        !this.state.isLoading &&
+        !this.isEqualTags(
+          this.formatTags(this.props.data),
+          this.state.prevInput,
+        )
+      ) {
+        this.fetchTimeout = setTimeout(() => {
+          this.runFetchImages();
+        }, 1000);
+      }
+    }
+
+    runFetchImages() {
+      const formattedTags: Array<ITagInput> = this.formatTags(this.props.data);
+      if (!formattedTags || formattedTags.length < 1) {
+        this.setState({
+          selectedImage: null,
+          images: [],
+        });
+        return;
+      }
+      this.setState({ isLoading: true, prevInput: formattedTags }, () => {
+        httpRequestJsonLocal<IImageServerResponse>({
+          abortSignal: this.abortController.signal,
+          method: "POST",
+          path: "/ai_image_suggestions/",
+          payload: {
+            service: "imatrics",
+            item: formattedTags,
+          },
+        })
+          .then((res) => {
+            if (this._mounted) {
+              this.setState({ fetchError: false });
+              try {
+                this.setState({
+                  selectedImage: res.result[0],
+                  images: res.result ?? [],
+                });
+              } catch {
+                this.setState({
+                  selectedImage: null,
+                  images: [],
+                });
+              }
+            }
+          })
+          .catch(() => {
+            this.setState({ fetchError: true });
+          })
+          .finally(() => this._mounted && this.setState({ isLoading: false }));
+      });
+    }
+
+    isEqualTags = (current: Array<ITagInput>, prev: Array<ITagInput>) =>
+      JSON.stringify(current) === JSON.stringify(prev)
+
+    formatTags = (concepts: IServerResponse) => {
+      let res: Array<ITagInput> = [];
+      for (const key in concepts) {
+        if (key === "subject") {
+          concepts[key]?.forEach((concept: ITagBase) => {
+            res.push({
+              title: concept.name,
+              type: "category",
+              pubStatus: true,
+              weight: 1,
+            });
+          });
+        } else if (
+          key === "organisation" ||
+          key === "person" ||
+          key === "event" ||
+          key === "place" ||
+          key === "object"
+        ) {
+          concepts[key]?.forEach((concept: ITagBase) => {
+            res.push({
+              title: concept.name,
+              type: key,
+              pubStatus: true,
+              weight: 1,
+            });
+          });
+        }
+      }
+      return res;
+    }
+
+    handleClickImage = (image: IImage) => {
+      this.setState({ selectedImage: image });
+    }
+
+    handleDragStart = (
+      event: React.DragEvent<HTMLDivElement>,
+      image: IImage | null,
+    ) => {
+      if (!image) {
+        return;
+      }
+      let item: IArticle;
+      try {
+        item = {
+          ...this.props.article,
```

Comment:
Hi, Tomas!

I noticed this and we have now fixed the second lint issue. Additionally, the payload has been adjusted according to the backend changes.

The users can also drag the images on the list since they have a similar attribute (onDragStart) to the selected image. However, you have a point that there is no indication that they are draggable. However, the indicator feature has not been demanded/explained by the stakeholder.

From the UX perspective, how may we provide an indicator of the drag feature without needing to fill up more space in the widget for future development?

Reply:
we allow external images, as long as it provides renditions with working urls it should be fine. ntb doesn't upload images in general, they use external images from scanpix